### PR TITLE
Add checkstyle plugin

### PR DESCRIPTION
**Pull request by juancruzgs**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -12,6 +12,22 @@ buildscript {
...
+    exclude '**/gen/**'
+    exclude '**/test/**'
+    exclude '**/R.java'
+    exclude '**/BuildConfig.java'
+
+    classpath = files()
+}
+
+// If checkstyle fails we can't generate a build
+preBuild.dependsOn('checkstyle')
```
Comment (julianfalcionelli) [inline app/build.gradle]: Doing this we don't need to add this custom gradle task to our Continuous Integration System, right ?

Comment (juancruzgs) [reply app/build.gradle]: Exactly, checkstyle always runs before a build

File: ci/checkstyle.xml [hunk 2]
```diff
@@ -0,0 +1,174 @@
+<?xml version="1.0"?>
```
Comment (julianfalcionelli) [inline ci/checkstyle.xml:1]: We have a section in our wiki with 2 suggested styles that we use, can we use the "Android Style with Improvements" style ? https://github.com/LateralView/android-base-project/wiki/Android-Studio-Configuration ?

Comment (juancruzgs) [reply ci/checkstyle.xml:1]: That style file applies to the IDE, and it has different format.
Try to reformat the code in this branch (Right click on main folder -> Reformat code) and let me know if you see any differences.

I took the checkstyle that Oracle is using and applied some changes

Comment (julianfalcionelli) [reply ci/checkstyle.xml:1]: Oh, I see, we need a specific xml file to set our checkstyle and the one that we use with the IDE is not compatible with this, right ? I am very sure that the checkstyle that you take from Oracle doesn't have the changes that I made on the Android recommended styles, like the life cycle arrangements, etc.

Comment (juancruzgs) [reply ci/checkstyle.xml:1]: They are not compatible.
The easiest way is to `Reformat` this code and see what changes.

**Review by julianfalcionelli — APPROVED** [review]

**Pull request merged**